logstash-core/api/lib/app/stats/memory_command.rb

```diff
@@ -0,0 +1,28 @@
+# encoding: utf-8
+require "app/command"
+require 'monitoring'
+
+class LogStash::Api::JvmMemoryCommand < LogStash::Api::Command
+
+  def run
+    memory = service.get(:jvm_memory_stats)
+    {
+      :heap => dump(memory[:heap].marshal_dump),
+      :non_heap => dump(memory[:non_heap].marshal_dump),
+      :pools => memory[:pools].marshal_dump.inject({}) do |acc, (type, hash)|
+                  acc[type] = dump(hash.marshal_dump)
+                  acc
+                end
+    }
+  end
+
+  private
+
+  def dump(hash)
+    hash.inject({}) do |h, (k,v)|
+      h[k] = v.value
+      h
+    end
+  end
+
```

Comment:
More than a ```to_s``` method, to have a marshal method that dump a structure as a hash, meaning without the require of a value method. I see this a very common use case for consumers, so yes, I would agree we should implemented in the ```metric_type``` or ```metric_store``` level.